src/Common/src/Interop/Unix/System.Native/Interop.Stat.Span.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        // Unix max paths are typically 1K or 4K, 256 should handle the majority of paths
+        // without putting too much pressure on the stack.
+        private const int StackBufferSize = 256;
+
+        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_Stat2", SetLastError = true)]
+        internal unsafe static extern int Stat_Byte(ref byte path, out FileStatus output);
+
+        internal unsafe static int Stat_Span(ReadOnlySpan<char> path, out FileStatus output)
+        {
+            byte* buffer = stackalloc byte[StackBufferSize];
+            var converter = new ValueUtf8Converter(new Span<byte>(buffer, StackBufferSize));
+            int result = Stat_Byte(ref MemoryMarshal.GetReference(converter.ConvertString(path)), out output);
+            converter.Clear();
+            return result;
+        }
+
+        [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_LStat2", SetLastError = true)]
+        internal static extern int LStat_Byte(ref byte path, out FileStatus output);
```

Comment:
Unusual to have two declarations of the same export (this is also in interop.stat.cs). Is the ref byte here a pointer?

Reply:
Unusual, but we do this in other places.  I'd remove the others, but they are in use in other libraries. I can make a follow-up issue to consider switching the others.

The ref is effectively a pointer. The marshaller will pin for you when you write it this way.